okhttp/src/main/kotlin/okhttp3/Cache.kt

```diff
@@ -228,9 +228,10 @@ class Cache internal constructor(
       return null
     }
 
-    if (requestMethod != "GET") {
-      // Don't cache non-GET responses. We're technically allowed to cache HEAD requests and some
-      // POST requests, but the complexity of doing so is high and the benefit is low.
+    if (requestMethod !in CACHEABLE_METHODS) {
```

Comment:
Rather annoyingly the previous check was very fast, and this check is quite slow. Checking presence in a set needs to allocate an iterator. I don’t like that this might matter?

Reply:
Doing uglier code is faster:
```suggestion
    if (requestMethod != "GET" && requestMethod != "QUERY") {
```